kong/api/endpoints.lua

```diff
@@ -321,8 +323,14 @@ local function get_collection_endpoint(schema, foreign_schema, foreign_field_nam
     end
 
     local foreign_key = self.params[foreign_schema.name]
-    local next_page = offset and fmt("/%s/%s/%s?offset=%s", foreign_schema.name,
-                                     foreign_key, schema.name, escape_uri(offset)) or null
+    local next_page = offset and fmt("/%s/%s/%s?offset=%s",
+                                     foreign_schema.path or
+                                     foreign_schema.name,
+                                     foreign_key,
+                                     schema.nested_path or
+                                     schema.path or
```

Comment:
I like that `name` is used as a fallback for the `foreign_schema` entry :+1: , but here I think `path` should not be used as a fallback to the `nested_path` entry (the reason becomes clearer when we consider the names I suggested below — a collection name should not be used as a fallback for an entity name).